### ISSUE TITLE
Can't decompile deodexed files on U11

### ISSUE THREAD

**Issue by jtothe4n** [anestisb/vdexExtractor#1]
I tried to decompile several files, framework.jar, telephony-common.jar, services.jar.

All fail with several errors that always look like:

>java -jar baksmali.jar d services.jar

>Error occurred while disassembling class _classname_ - skipping class
org.jf.util.ExceptionWithContext: The last instruction in method _methodname_ is truncated

When I deodex with baksmali, it works fine.

Other files I tried, like SystemUI.apk work fine.

[services.zip](https://github.com/anestisb/vdexExtractor/files/1451371/services.zip)

**Comment by anestisb** [comment]
Thanks for reporting. Will investigate.

**Comment by jtothe4n** [comment]
Seems using the -u flag (enable unquicken bytecode decompiler) fixes the issue. Whatever it is doing (-:

**Comment by anestisb** [comment]
If you don't invoke with `-u | --unquicken` it just simply grabs the Dex file from Vdex without decompiling (deodex in simpler terms). So you were dumping an optimized version thus the issues you had, and I was struggling to replicate.

**Comment by jtothe4n** [comment]
Thanks for the explanation, would have expect that being the default behavior.
Cheers, Jan.
